File: HW10/inLevel.java
import java.util.List;
import java.util.ArrayList;
import java.util.Queue;
import java.util.LinkedList;

public class inLevel
{
    public static List<List<Integer>> levelOrder(TreeNode root)
    {
        List<List<Integer>> tree = new ArrayList<>();

        if(root == null)
        {
            return tree;
        }

        Queue<TreeNode> holder = new LinkedList<>();
        holder.add(root);

        while(!holder.isEmpty())
        {
            int size = holder.size();
            List<Integer> levels = new LinkedList<Integer>();

            for(int i = 0; i < size; i++)
            {
                TreeNode current = holder.remove();
                levels.add(current.val);

                if(current.left != null)
                {
                    holder.add(current.left);
                }
                if(current.right != null)
                {
                    holder.add(current.right);
                }
            }

            tree.add(levels);
        }

    return tree;   
    }
    public static void main(String[] args)
    {
        TreeNode root = new TreeNode(4);
        TreeNode three = root.left = new TreeNode(3);
        TreeNode eight = root.right = new TreeNode(8);
        TreeNode one = root.left.left = new TreeNode(1);
        TreeNode five = root.right.left = new TreeNode(5);
        TreeNode nine = root.right.right = new TreeNode(9);

        List<List<Integer>> holder = levelOrder(root);

        
        for(List<Integer> level : holder)
        {
                System.out.println(level);
        }
    }
}

File: HW10/inLevel.py
from typing import List, Optional
from collections import deque

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def levelOrder(root: Optional[TreeNode]) -> List[List[int]]:
    tree = []
    if root is None:
        return tree
    
    holder = deque()
    holder.append(root)

    while holder:
        size = len(holder)
        levels = []

        for _ in range(size):
            current = holder.popleft()
            levels.append(current.val)

            if current.left:
                holder.append(current.left)
            if current.right:
                holder.append(current.right)

        tree.append(levels)

    return tree

def main():
    root = TreeNode(4)
    root.left = TreeNode(3)
    root.right = TreeNode(8)
    root.left.left = TreeNode(1)
    root.right.left = TreeNode(5)
    root.right.right = TreeNode(9)

    holder = levelOrder(root)

    for level in holder:
        print(level)

if __name__ == "__main__":
    main()
